src/viur/shop/types/price.py

```diff
@@ -124,13 +124,17 @@ def choose_best_discount_set(self) -> tuple[float, list[SkeletonInstance]]:
                 logger.info(f"Not suitable for combinables")
                 continue
         all_permutations.append(combinables)
-
         best_price = self.retail
         best_discounts = None
         for permutation in all_permutations:
             price = self.retail  # start always from the retail price
             for discount in permutation:
-                price = self.apply_discount(discount, price)
+                # only add if ApplicationDomain.ARTICLE
+                if any(
+                    [condition["dest"]["application_domain"] == ApplicationDomain.ARTICLE
+                     for condition in discount["condition"]]
+                ):
```

Comment:
Please use a generator comprehension
```suggestion
                if any(
                    condition["dest"]["application_domain"] == ApplicationDomain.ARTICLE
                    for condition in discount["condition"]
                ):
```